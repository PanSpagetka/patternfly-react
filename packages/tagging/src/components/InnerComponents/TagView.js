import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'patternfly-react';
import TagCategory from './TagCategory';

class TagView extends React.Component {
  generateTagCategories = tag => (
    <TagCategory
      key={tag.id}
      tagCategory={{ id: tag.id, description: tag.description }}
      tagValues={tag.tagValues}
      onTagDeleteClick={this.props.onTagDeleteClick}
    />
  );

  render() {
    const assignedTags = [...this.props.assignedTags];
    return (
      <React.Fragment>
        <Row>
          <Col lg={12}>
            <h2>{this.props.header}</h2>
          </Col>
        </Row>
        <Row>
          <Col lg={12}>
            <ul className="list-inline">
              {assignedTags
                .sort((a, b) => (a.description < b.description ? -1 : 1))
                .map(this.generateTagCategories)}
            </ul>
          </Col>
        </Row>
      </React.Fragment>
    );
  }
}
TagView.propTypes = {
  assignedTags: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
      tagValues: PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.number.isRequired,
          description: PropTypes.string.isRequired
        }).isRequired
      ).isRequired
    })
  ).isRequired,
  onTagDeleteClick: PropTypes.func.isRequired,
  header: PropTypes.string
};

TagView.defaultProps = {
  header: 'Assigned tags'
};

export default TagView;
