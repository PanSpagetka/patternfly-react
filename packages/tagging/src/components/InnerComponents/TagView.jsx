import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'patternfly-react';
import TagCategory from './TagCategory';

class TagView extends React.Component {
  generateTagCategories = tag =>
    (<TagCategory
      key={tag.tagCategory.id}
      tagCategory={tag.tagCategory}
      tagValues={tag.tagValues}
      onTagDeleteClick={this.props.onTagDeleteClick}
    />);

  render() {
    let assignedTags = [...this.props.assignedTags];
    return (
      <React.Fragment>
        <Row><Col lg={12}><h2>{this.props.header}</h2></Col></Row>
        <Row>
          <Col lg={12}>
            <ul className="list-inline">
              {assignedTags.sort((a, b) => a.tagCategory.description < b.tagCategory.description ? -1 : 1).map(this.generateTagCategories)}
            </ul>
          </Col>
        </Row>
      </React.Fragment>
    );
  }
}

TagView.propTypes = {
  assignedTags: PropTypes.arrayOf(PropTypes.object),
  onTagDeleteClick: PropTypes.func.isRequired,
  header: PropTypes.string,
};

TagView.defaultProps = {
  header: 'Assigned tags',
};

export default TagView;
