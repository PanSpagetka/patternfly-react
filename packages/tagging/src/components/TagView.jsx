import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'patternfly-react';
import TagCategory from './TagCategory';

class TagView extends React.Component {
  generateTagCategories = tag =>
    (<TagCategory
      key={tag.tagCategory.id}
      tagCategory={tag.tagCategory}
      tagValues={tag.tagValues}
      onTagDeleteClick={this.props.onTagDeleteClick}
    />);

  render() {
    return (
      <React.Fragment>
        <Row><Col lg={12}><h2>{this.props.header}</h2></Col></Row>
        <Row>
          <Col lg={12}>
            <ul className="list-inline">
              {this.props.assignedTags.map(this.generateTagCategories)}
            </ul>
          </Col>
        </Row>
      </React.Fragment>
    );
  }
}

TagView.propTypes = {
  assignedTags: PropTypes.arrayOf(PropTypes.object),
  onTagDeleteClick: PropTypes.func.isRequired,
  header: PropTypes.string,
};

TagView.defaultProps = {
  header: 'Assigned tags',
};

export default TagView;
