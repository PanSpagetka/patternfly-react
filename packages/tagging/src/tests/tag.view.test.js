import React from 'react';
import TagView from '../components/TagView';
import renderer from 'react-test-renderer';

const assignedTags = [
  {
    tagCategory: { description: 'Name', id: 1 },
    tagValues: [{ description: 'Pepa', id: 11 }]
  }
];
function onDelete(x) {
  return x;
}
describe('Tag view', () => {
  it('match snapshot', () => {
    const component = renderer.create(
      <TagView assignedTags={assignedTags} onTagDeleteClick={onDelete} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
