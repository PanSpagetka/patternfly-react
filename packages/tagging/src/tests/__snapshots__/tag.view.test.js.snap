// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag view match snapshot 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <h2>
        Assigned tags
      </h2>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <ul
        className="list-inline"
      >
        <ul
          className="tag-category list-inline"
        >
          <div
            className="category-label"
          >
            Name
          </div>
          <li
            className="tag"
          >
            <span
              className="label label-primary"
              style={
                Object {
                  "color": "defaultColor",
                }
              }
            >
              Pepa
              <a
                className="pf-remove-button"
                href="#"
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="pficon pficon-close"
                />
                <span
                  className="sr-only"
                >
                  Remove
                </span>
              </a>
            </span>
          </li>
        </ul>
      </ul>
    </div>
  </div>,
]
`;
